Source/WebCore/Modules/mediasource/MediaSource.cpp

```diff
@@ -1337,12 +1337,11 @@ void MediaSource::notifyElementUpdateMediaState() const
 void MediaSource::ensureWeakOnHTMLMediaElementContext(Function<void(HTMLMediaElement&)>&& task) const
 {
     ensureOnMainThread([weakMediaElement = m_mediaElement, task = WTFMove(task)]() mutable {
-        if (RefPtr mediaElement = weakMediaElement.get())
+        if (auto mediaElement = weakMediaElement.get())
```

Comment:
This goes against our smart pointer guidelines. r-.

Please use `RefPtrAllowingPartiallyDestroyed<MediaElement>`.

Reply:
Thanks

Reply:
Where are the guidelines?

Reply:
https://github.com/WebKit/WebKit/wiki/Smart-Pointer-Usage-Guidelines